src/main/scala/nl.knaw.dans.easy.dd2d/DepositSorter.scala

```diff
@@ -0,0 +1,106 @@
+
+  private case class DepositsSortInfo(name: String, timeStamp: LocalDateTime, isVersionOf: Option[String], depositTask: DepositIngestTask)
+  private type DepositIngestTaskList = List[DepositIngestTask]
+  private type SortInfoList = List[DepositsSortInfo]
+  private type VersionMap = Map[String, List[DepositsSortInfo]]
+  private val BAG_INFO_FILE = "bag-info.txt"
+  private val IS_VERSION_OF = "Is-Version-Of"
+  private val CREATED = "Created"
+
+  /**
+   * sorts the (DepositIngest)Tasks by dataset and per dataset by version
+   *
+   * @param tasks list of DepositIngestTasks
+   * @return the sorted list of deposit directories
+   */
+  override def sort(tasks: List[Task[Deposit]]): List[Task[Deposit]] = {
+    val sortInfoList = tasks.map(d => getDepositSortInfo(d.asInstanceOf[DepositIngestTask]))
+    val groupedDeposits = groupVersions(sortInfoList)
+    sortVersions(groupedDeposits)
+  }
+
+  /**
+   * creates a DepositsSortInfo object that contains necessary sorting information
+   *
+   * @param depositIngestTask
+   * @return DepositsSortInfo
+   */
+  private def getDepositSortInfo(depositIngestTask: DepositIngestTask): DepositsSortInfo = {
+    val bagInfoFile = depositIngestTask.deposit.dir.list(_.isRegularFile, 2).filter(_.name == BAG_INFO_FILE).toList.head
+    val properties = new Properties()
+    properties.load(new FileInputStream(bagInfoFile.pathAsString));
+    val timeStamp = parseTimestamp(properties.getProperty(CREATED))
+    val isVersionOf: Option[String] = properties.getProperty(IS_VERSION_OF) match {
+      case null => None
+      case uuid: String => Some(uuid)
+    }
+    DepositsSortInfo(depositIngestTask.deposit.dir.name, timeStamp, isVersionOf, depositIngestTask)
+  }
+
+  private def groupVersions(depositsSortInfoList: SortInfoList): VersionMap = {
+    val firstVersions = depositsSortInfoList
+      .filter(_.isVersionOf.isEmpty)
+      .map(v => v.name -> List(v)).toMap
+
+    val laterVersions =
+      depositsSortInfoList
+        .filter(_.isVersionOf.isDefined)
+        .groupBy(_.isVersionOf.get)
+
+    val laterVersionsUpdated = removeDatasetsWithoutFirstVersion(firstVersions, laterVersions)
+
+    (firstVersions.keySet ++ laterVersionsUpdated.keySet)
+      .map(k => k -> (firstVersions(k) ++ laterVersionsUpdated(k)))
+      .toMap
+  }
+
+  private def sortVersions(groupedDeposits: VersionMap): DepositIngestTaskList = {
+    groupedDeposits.mapValues(k => sortByTimestamp(k))
+      .flatMap(_._2)
+      .map(_.depositTask)
+      .toList
+  }
+
+  private def removeDatasetsWithoutFirstVersion(firstVersions: VersionMap, laterVersions: VersionMap): VersionMap = {
+    val datasetsWithoutFirstVersions = for (k <- laterVersions if !firstVersions.keySet.contains(k._1)) yield k
+    datasetsWithoutFirstVersions.foreach(v => logger.error(s"No first version was found for dataset ${ v._1 }. The dataset was not imported into Dataverse"))
```

Comment:
in general, avoid `._1` and `._2` and find something more readable  
```suggestion
    datasetsWithoutFirstVersions.foreach{ case (k,v) => 
      logger.error(s"No first version was found for dataset $ v. The dataset was not imported into Dataverse")
    }
```